### PR TITLE
refactor: simplify can_cast_types code.

### PR DESCRIPTION
**Pull request by jackwener**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

# What changes are included in this PR?

simplify can_cast_types code

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-cast/src/cast.rs [hunk 1]
```diff
@@ -149,114 +153,54 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
         (Null | Int8 | Int16 | Int32 | Int64 | Float32 | Float64, Decimal128(_, _)) |
         (Null | Int8 | Int16 | Int32 | Int64 | Float32 | Float64, Decimal256(_, _)) |
         // decimal to unsigned numeric
-        (Decimal128(_, _), UInt8 | UInt16 | UInt32 | UInt64) |
-        (Decimal256(_, _), UInt8 | UInt16 | UInt32 | UInt64) |
+        (Decimal128(_, _)|Decimal256(_, _), UInt8 | UInt16 | UInt32 | UInt64) |
         // decimal to signed numeric
-        (Decimal128(_, _), Null | Int8 | Int16 | Int32 | Int64 | Float32 | Float64) |
-        (Decimal256(_, _), Null | Int8 | Int16 | Int32 | Int64 | Float32 | Float64) => true,
+        (Decimal128(_, _)|Decimal256(_, _), Null | Int8 | Int16 | Int32 | Int64 | Float32 | Float64) => true,
```
Comment (viirya) [inline arrow-cast/src/cast.rs:158]:
```suggestion
        (Decimal128(_, _) | Decimal256(_, _), Null | Int8 | Int16 | Int32 | Int64 | Float32 | Float64) => true,
```

File: arrow-cast/src/cast.rs [hunk 2]
```diff
@@ -149,114 +153,54 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
...
+        (Decimal128(_, _)|Decimal256(_, _), Utf8 | LargeUtf8) => true,
         // Utf8 to decimal
-        (Utf8 | LargeUtf8, Decimal128(_, _)) => true,
-        (Utf8 | LargeUtf8, Decimal256(_, _)) => true,
-        (Decimal128(_, _), _) => false,
-        (_, Decimal128(_, _)) => false,
-        (Decimal256(_, _), _) => false,
-        (_, Decimal256(_, _)) => false,
+        (Utf8 | LargeUtf8, Decimal128(_, _)|Decimal256(_, _)) => true,
+        (Decimal128(_, _)|Decimal256(_, _), _) => false,
```
Comment (viirya) [inline arrow-cast/src/cast.rs:163]:
```suggestion
        (Decimal128(_, _) | Decimal256(_, _), _) => false,
```

File: arrow-cast/src/cast.rs [hunk 3]
```diff
@@ -149,114 +153,54 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
...
-        (Decimal256(_, _), Utf8 | LargeUtf8) => true,
+        (Decimal128(_, _)|Decimal256(_, _), Utf8 | LargeUtf8) => true,
         // Utf8 to decimal
-        (Utf8 | LargeUtf8, Decimal128(_, _)) => true,
-        (Utf8 | LargeUtf8, Decimal256(_, _)) => true,
-        (Decimal128(_, _), _) => false,
-        (_, Decimal128(_, _)) => false,
-        (Decimal256(_, _), _) => false,
-        (_, Decimal256(_, _)) => false,
+        (Utf8 | LargeUtf8, Decimal128(_, _)|Decimal256(_, _)) => true,
```
Comment (viirya) [inline arrow-cast/src/cast.rs:162]:
```suggestion
        (Utf8 | LargeUtf8, Decimal128(_, _) | Decimal256(_, _)) => true,
```

File: arrow-cast/src/cast.rs [hunk 4]
```diff
@@ -149,114 +153,54 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
...
+        (Decimal128(_, _)|Decimal256(_, _), _) => false,
+        (_, Decimal128(_, _)|Decimal256(_, _)) => false,
```
Comment (viirya) [inline arrow-cast/src/cast.rs:164]:
```suggestion
        (_, Decimal128(_, _) | Decimal256(_, _)) => false,
```

File: arrow-cast/src/cast.rs [hunk 5]
```diff
@@ -149,114 +153,54 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
...
-
-        (
-            Float32,
-            UInt8 | UInt16 | UInt32 | UInt64 | Int8 | Int16 | Int32 | Int64 | Float64,
-        ) => true,
-
-        (
-            Float64,
-            UInt8 | UInt16 | UInt32 | UInt64 | Int8 | Int16 | Int32 | Int64 | Float32,
+            UInt8|UInt16 | UInt32 | UInt64 | Int8 | Int16 | Int32 | Int64 | Float32 | Float64,
```
Comment (viirya) [inline arrow-cast/src/cast.rs:202]:
```suggestion
            UInt8 | UInt16 | UInt32 | UInt64 | Int8 | Int16 | Int32 | Int64 | Float32 | Float64,
```

File: arrow-cast/src/cast.rs [hunk 6]
```diff
@@ -149,114 +153,54 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
...
+            UInt8|UInt16 | UInt32 | UInt64 | Int8 | Int16 | Int32 | Int64 | Float32 | Float64,
```
Comment (viirya) [inline arrow-cast/src/cast.rs:203]:
```suggestion
            UInt8 | UInt16 | UInt32 | UInt64 | Int8 | Int16 | Int32 | Int64 | Float32 | Float64,
```

File: arrow-cast/src/cast.rs [hunk 7]
```diff
@@ -149,114 +153,54 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
...
         // Utf8 to decimal
-        (Utf8 | LargeUtf8, Decimal128(_, _)) => true,
-        (Utf8 | LargeUtf8, Decimal256(_, _)) => true,
-        (Decimal128(_, _), _) => false,
-        (_, Decimal128(_, _)) => false,
-        (Decimal256(_, _), _) => false,
-        (_, Decimal256(_, _)) => false,
+        (Utf8 | LargeUtf8, Decimal128(_, _) | Decimal256(_, _)) => true,
+        (Decimal128(_, _) | Decimal256(_, _), _) => false,
+        (_, Decimal128(_, _) | Decimal256(_, _)) => false,
```
Comment (viirya) [inline arrow-cast/src/cast.rs:164]: Hm, is this same as before?

File: arrow-cast/src/cast.rs [hunk 8]
```diff
@@ -149,114 +153,54 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
...
-        (Decimal128(_, _), Utf8 | LargeUtf8) => true,
-        (Decimal256(_, _), Utf8 | LargeUtf8) => true,
+        (Decimal128(_, _) | Decimal256(_, _), Utf8 | LargeUtf8) => true,
         // Utf8 to decimal
-        (Utf8 | LargeUtf8, Decimal128(_, _)) => true,
-        (Utf8 | LargeUtf8, Decimal256(_, _)) => true,
-        (Decimal128(_, _), _) => false,
-        (_, Decimal128(_, _)) => false,
-        (Decimal256(_, _), _) => false,
-        (_, Decimal256(_, _)) => false,
```
Comment (jackwener) [inline arrow-cast/src/cast.rs:166]: > Hm, is this same as before?

@viirya  Yes, I just merge them together

**Review by viirya — APPROVED** [review]

**Review by viirya — APPROVED** [review]

File: arrow-cast/src/cast.rs (hunk 8 above)
Comment (viirya) [reply arrow-cast/src/cast.rs:166]: Okay.

**Comment by jackwener** [comment]
thanks @viirya

**Pull request merged**